articles/how-to-use-an-api-in-a-vuejs-project/index.md

```diff
@@ -0,0 +1,345 @@
+```
+
+**How to make an API request and display data using Axios.**
+
+We will make the API request using the `get` method. A get method is used to request data from a resource, the resource in this context is our API.  We want this API request running asynchronously therefore, we use a promise based function with keywords `async` and `await`. However, we also need to test for errors, this is done with the `try` method and the errors that is detected is handled with the `catch` method.
+
+Copy the code below to your `App.vue` file
+
+```JavaScript
+<template></template>
+
+<script>
+export default {
+data()  {
+posts: [ ]
+
+},
+
+methods: {
+async getData () {
+     try {
+              const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+              // JSON responses are automatically parsed.
+              this.posts = response.data
+      }
+      catch (error) {
+          console.log(error);
+      }
+
+     }
+   }
+}
+</script>
+```
+
+After requesting for data from the API, you will need to call it on a lifecycle hook. Here we will use the `created()` lifecycle hook, this is because we will be able to access reactive data and events that are active with the `created` hook.
+
+
+    <template></template>
+    
+    <script>
+    export default {
+    data()  {
+    posts: [ ]
+    
+    },
+    
+    methods: {
+    async getData () {
+         try {
+                  const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+                 // JSON responses are automatically parsed.
+                  this.posts = response.data
+                  console.log(posts)
+               }
+              catch (error) {
+                 console.log(error);
+             }
+    
+           }
+       },
+    
+    
+    created() {
+        this.getData();
+      },
+    }
+    </script>
+
+We can now display the data in the template by looping through the posts using [v-for](https://vuejs.org/v2/guide/list.html) directive
+
+
+    <template>
+         <div>
+             <div v-for="post in posts">
+                 <h2>{{ post.title }}</h2>
+                  <p>{{ post.body }}</p>
+             </div>
+          </div>
+    </template>
+    
+    <script>
+    export default {
+    data()  {
+      posts: [ ]
+    },
+    
+    methods: {
+    async getData () {
+         try {
+                  const response = await this.$http.get('http://jsonplaceholder.typicode.com/posts');
+                  this.posts = response.data
+               }
+              catch (error) {
+                 console.log(error);
+             }
+    
+           }
+       },
+    
+    
+    created() {
+        this.getData();
+      },
+    }
+    </script>
+
+
+
+## Using Fetch API method
+
+Fetch API is a powerful and flexible method of flexible APIs. It provides a global `fetch()` method that provides an easy, logical way to fetch resources asynchronously across the network.
```

Comment:
Provide the link to the Fetch API [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)